skill-tree /src/test/java/com/RDS/skilltree/SkillsIntegrationTests.java

```diff
@@ -1,137 +1,250 @@
+
+        response.then()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("name", equalTo("Java"))
+                .body("type", equalTo("ATOMIC"));
+    }
+
+    @Test
+    @DisplayName("Return 404, if skill given skill name is not found")
+    public void testAPIReturns404_OnSkillGivenSkillNameNotFound() {
+        Response response = given()
+                .pathParam("name", "Go")
+                .get("/v1/skills/name/{name}");
+
 
         response.then()
                 .statusCode(404)
-                .body("code", equalTo(404))
-                .body("message",equalTo("No skills associated with the user"));
+                .body("message", equalTo("Skill not found with the given name"));
+    }
 
+    @Test
+    @DisplayName("Return 400, if createdBy is not passed for Skill creation")
+    public void testAPIReturns400_OnCreatedByNotPassedForSKillCreation() {
+        SkillDRO skillDRO = SkillDRO.builder()
+                .name("Go")
+                .type(SkillType.ATOMIC)
+                .build();
+
+        Response response = given()
+                .contentType("application/json")
+                .body(skillDRO)
+                .post("/v1/skills/");
+
+        response.then()
+                .statusCode(400)
+                .body("error", equalTo("Bad Request"));
     }
 
     @Test
-    @Disabled
-    @DisplayName("Return users having skill given skill name")
-    public void testAPIReturns200_OnSkillGivenSkillName() {
+    @DisplayName("Return 400, if type is not passed for Skill creation")
+    public void testAPIReturns400_OnTypeNotPassedForSkillCreation() {
+        SkillDRO skillDRO = SkillDRO.builder()
+                .name("Go")
+                .createdBy(user.getId())
+                .build();
+
         Response response = given()
-                .get("/skills/{skillName}");
+                .contentType("application/json")
+                .body(skillDRO)
+                .post("/v1/skills/");
 
-        /*
+        response.then()
+                .statusCode(400)
+                .body("error", equalTo("Bad Request")); //Todo change this on introducing the global exception handling
+    }
+
+    @Test
+    @DisplayName("Return 400, if name is not passed for Skill creation")
+    public void testAPIReturns400_OnNameNotPassedForSkillCreation() {
+        SkillDRO skillDRO = SkillDRO.builder()
+                .type(SkillType.ATOMIC)
+                .createdBy(user.getId())
+                .build();
 
-        [
-          {
-            "skillName": "Java",
-            "userName": "John doe"
-          }
-        ]
-         */
+        Response response = given()
+                .contentType("application/json")
+                .body(skillDRO)
+                .post("/v1/skills/");
 
         response.then()
-                .statusCode(200)
+                .statusCode(400)
+                .body("error", equalTo("Bad Request"));
```

Comment:
can we also provide a message so that the client has a clue what went wrong

Reply:
The custom exception handling needs to be done, will look into that

Reply:
same as above